indico/modules/events/registration/controllers/management/regforms.py

```diff
@@ -252,6 +255,37 @@ def _process(self):
                                                     regform=self.regform)
 
 
+class RHRegistrationFormNotificationPreview(RHManageRegFormBase):
+    """Preview registration emails."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.commit = False
+
+    @no_autoflush
+    def _process(self):
+        message = request.form.get('text', '')
+        state = RegistrationState.get(request.form.get('state', RegistrationState.pending.name))
+        if state == RegistrationState.pending:
+            self.regform.message_pending = message
+        elif state == RegistrationState.unpaid:
+            self.regform.message_unpaid = message
+        elif state == RegistrationState.complete:
+            self.regform.message_complete = message
+        else:
+            raise ValidationError(_('Invalid state'))
+        mock_registration = Registration(state=state, registration_form=self.regform, currency='USD',
+                                         email='test@email.com', first_name=_('{FIRST_NAME}'),
+                                         last_name=_('{LAST_NAME}'), checked_in=True, friendly_id=-1,
```

Comment:
I don't think translating placeholders is a good idea. let's maybe use a dummy name like we do in `create_mock_abstract`? Also, depending on how much you need from the real Registration object, maybe best to use a namedtuple as a mock to avoid having to care about things ending up in the DB altogether :)

Reply:
There are many computed properties in that object. Mocking all that would be a hassle, but as you mention it has the disadvantage of ghost insertions in the database.

I did try on the first place with named tuples, but ended up discarding it, as it was a mess to compute all properties. It'd also be less maintainable in the future in case the database object changes, so I thought that as long as we are 100% there are no hidden insertions we should be fine with raw objects.

Reply:
Yes, I was afraid that of those properties are needed...
But if you use the real model, then it might be nice to also add some of the default Personal Data fields so the email preview looks less broken? As long as you only link them to the Registration instance but not to an event or any other that's already in the DB, it should never get flushed (even if there wasn't a `no_autoflush`).

Reply:
I placed some dummy data, and confirm that no data is ever flushed.